Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -768,34 +768,66 @@ extension Tensor where Scalar: TensorFlowFloatingPoint {
 // Padding
 //===------------------------------------------------------------------------------------------===//
 
+
 public extension Tensor where Scalar: Numeric {
-    /// Returns a padded tensor according to the specified padding sizes.
+    /// Padding modes:
+    ///  * `constant` - pad with constant value.
+    ///  * `reflect` - mirror values along padding dimensions, excluding the edge value.
+    ///  * `symmetric` - mirror values along padding dimensions, including the edge value.
+    enum PaddingMode {
+        case constant(Scalar)
+        case reflect
+        case symmetric
+    }
+
+    /// Returns a tensor padded with constant according to the specified padding sizes.
     @inlinable
-    @differentiable(wrt: self, vjp: _vjpPadded(forSizes:with:) where Scalar: TensorFlowFloatingPoint)
+    @differentiable(wrt: self where Scalar: TensorFlowFloatingPoint)
     func padded(forSizes sizes: [(before: Int, after: Int)], with value: Scalar = 0) -> Tensor {
+        return self.padded(forSizes: sizes, with: .constant(value))
+    }
+
+    /// Returns a padded tensor according to the specified padding sizes and mode.
+    @inlinable
+    @differentiable(wrt: self, vjp: _vjpPaddedWithMode(forSizes:with:) where Scalar: TensorFlowFloatingPoint)
+    func padded(forSizes sizes: [(before: Int, after: Int)], with mode: PaddingMode) -> Tensor {
         let paddings = Tensor<Int32>(
             shape: [sizes.count, 2],
             scalars: sizes.flatMap { [Int32($0.before), Int32($0.after)] })
-        return Raw.padV2(self, paddings: paddings, constantValues: Tensor(value))
+        switch mode {
+        case .constant(let constantValue):
+            return Raw.padV2(self, paddings: paddings, constantValues: Tensor(constantValue))
+        case .reflect:
+            return Raw.mirrorPad(self, paddings: paddings, mode: Raw.Mode5.reflect)
```

Comment:
Omit explicit base type information when there's a contextual type.
```suggestion
            return Raw.mirrorPad(self, paddings: paddings, mode: .reflect)
```